src/commands/deleteCommand.ts

```diff
@@ -0,0 +1,81 @@
+import { setTimeout as wait } from "node:timers/promises";
+import { Command } from "@yuudachi/framework";
+import type { InteractionParam, ArgsParam, CommandMethod } from "@yuudachi/framework/types";
+import { PermissionFlagsBits, time, TimestampStyles } from "discord.js";
+import type { DeleteCommandContextCommand } from "../interactions/context/deleteCommand.js";
+import { GITHUB_THREAD_NAME } from "../util/constants.js";
+
+export default class extends Command<typeof DeleteCommandContextCommand> {
+	public constructor() {
+		super(["Delete Command"]);
+	}
+
+	private matchChannelName(name: string): boolean {
+		return name === GITHUB_THREAD_NAME;
+	}
+
+	public override async messageContext(
+		interaction: InteractionParam<CommandMethod.MessageContext>,
+		args: ArgsParam<typeof DeleteCommandContextCommand>,
+	): Promise<void> {
+		if (args.message.interaction) {
+			if (
+				args.message.interaction.user.id !== interaction.user.id ||
+				!interaction.memberPermissions.has(PermissionFlagsBits.ManageMessages)
```

Comment:
manage messages should not be required, if the author of the command is the author of the context menu